src/packages/ganache/tests/types.test.ts

```diff
@@ -0,0 +1,183 @@
+import { Ethereum } from "../"; // <- same as `from "ganache"`
+
+//#region type helpers
+/**
+ * Converts a union into an intersection, as long as the types don't collide:
+ *
+ * ```typescript
+ * // this is fine:
+ * UnionToIntersection<{prop: string} | {other: boolean}> // `{prop: string;} & {other: boolean;}`
+ * ```
+ * ```
+ * // returns `never`:
+ * UnionToIntersection<{prop: string} | {prop: boolean}> // never
+ * ```
+ */
+type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) extends (
+  k: infer I
+) => void
+  ? I
+  : never;
+
+/**
+ * Returns only the last item in a Tuple
+ */
+type LastOf<T> = UnionToIntersection<
+  T extends any ? () => T : never
+> extends () => infer R
+  ? R
+  : never;
+
+/**
+ * Pushes a new value, V, on to the Tuple, T
+ **/
+type Push<T extends any[], V> = [...T, V];
+
+/**
+ * `TuplifyUnion` splits booleans into [true, false], but we really want to represent that as just `boolean`
+ */
+type NormalizeBoolean<MaybeBoolean> = MaybeBoolean extends
+  | [true, false]
+  | [false, true]
+  ? [boolean]
+  : MaybeBoolean;
+
+/**
+ * Convert a Union into Tuple a union.
```

Comment:
```suggestion
 * Convert a Union into a Tuple union.
```
Did I get that right?

Reply:
> Convert a Union into a Tuple

was the right answer. haha. fixed!